apps/04TowersOfHanoi.js

```diff
@@ -19,23 +19,59 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  var starting = stacks[startStack];
+  var startPiece = starting.pop();
+  var end = stacks[endStack];
+  var ending = null;
+  ending = end.push(startPiece);
 }
 
-function isLegal() {
-  // Your code here
+function isLegal(startStack, endStack) {
+  var starting = stacks[startStack];
+  var end = stacks[endStack];
+  var ending = null;
+  if (starting[starting.length - 1] > end[end.length - 1]) {
```

Comment:
What if start/end arrays are empty? What values will this comparison be comparing?